api/v1alpha1/releasestrategy_types.go

```diff
@@ -22,15 +22,31 @@ import (
 
 // ReleaseStrategySpec defines the desired state of ReleaseStrategy
 type ReleaseStrategySpec struct {
-	// Release Tekton Pipeline to execute
+	// Release Tekton Pipeline to execute; used with bundle resolver
 	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
-	// +required
-	Pipeline string `json:"pipeline"`
+	// +optional
+	Pipeline string `json:"pipeline,omitempty"`
 
-	// Bundle is a reference to the Tekton bundle where to find the pipeline
+	// Bundle is a reference to the Tekton bundle where to find the pipeline;
+	// used with bundle resolver
 	// +optional
 	Bundle string `json:"bundle,omitempty"`
 
+	// Git repo URL where the Release Tekton Pipeline to execute is located;
+	// used with git resolver
+	// +optional
+	GitUrl string `json:"gitUrl,omitempty"`
+
+	// Revision of git repo where the Release Tekton Pipeline to execute is located;
+	// used with git resolver
+	// +optional
+	GitRevision string `json:"gitRevision,omitempty"`
+
+	// Relative path in git repo where the Release Tekton Pipeline to execute is located;
+	// used with git resolver
+	// +optional
+	GitPathInRepo string `json:"gitPathInRepo,omitempty"`
```

Comment:
This should be a struct. Something like `Resolver` which would have a `BundleReference` or a `GitRefeference`. I'd make both of them structs as well so for example, the `GitReference` one would have the fields `Url`, `Revision` and `PathInRepo` while the `BundleReference` would just have a `Bundle` field.

The change to support dev will come with plenty of changes, so I don't really mind doing that here. @johnbieren?

Reply:
I'm not sure I follow what the question is

Reply:
@davidmogar ok, I will do that. But in that case it will be a breaking change, so it definitely needs to be done together with the controller change.

> the BundleReference would just have a Bundle field.

You also need a name for the bundle resolver. (Currently the `Pipeline` field.)

Reply:
> I'm not sure I follow what the question is

Whether we should do it here or wait for the change supporting dev pipelines.

Reply:
It doesn't matter to me whether it is done all in this PR after https://github.com/redhat-appstudio/release-service/pull/246 merges or in a followup. Whatever is easiest